port/linux/golioth_sys_linux.c

```diff
@@ -47,6 +47,54 @@ uint64_t golioth_sys_now_ms(void)
     return now_ms - start_ms;
 }
 
+/*--------------------------------------------------
+ * Mutexes
+ *------------------------------------------------*/
+
+golioth_sys_mutex_t golioth_sys_mutex_create(void)
+{
+    pthread_mutex_t *mutex = golioth_sys_malloc(sizeof(pthread_mutex_t));
+    int err = pthread_mutex_init(mutex, NULL);
+    if (err)
+    {
+        assert(false);
+        return NULL;
+    }
```

Comment:
We can just assert that `err == 0` directly. Also, if there is an error, we should free the mutex before returning `NULL`, or we will leak memory.